lib/galaxy/managers/users.py

```diff
@@ -613,6 +613,8 @@ def send_reset_email(self, trans, payload, **kwd):
 
     def get_reset_token(self, trans, email):
         reset_user = get_user_by_email(trans.sa_session, email, self.app.model.User)
+        if reset_user.deleted:
+            return None, None
         if not reset_user and email != email.lower():
             reset_user = self._get_user_by_email_case_insensitive(trans.sa_session, email)
         if reset_user:
```

Comment:
Or maybe the check above could be removed and we change line 620 to:
```suggestion
        if reset_user and not reset_user.deleted:
```

Reply:
In fact, this is the correct version: the previous one would have been a bug (with or without my previous suggestion), because if we fail to retrieve the user the first time, we check again using the case-insensitive method - so we should check for `.deleted` *after* that second attempt. Which is what this version does.